src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,257 @@ begin
+  simp only [(fin_succ_equiv R n).map_add, polynomial.coeff_add, coeff_add, hp, hq],
+  apply fin_succ_equiv_coeff_coeff_case_p_X,
+end
+
+lemma eval_eq_eval_mv_eval' {n : ℕ} {R : Type u} [comm_semiring R] (s : fin n → R) (y : R)
+  (f : mv_polynomial (fin (n + 1)) R) : eval (fin.cons y s : fin (n + 1) → R) f
+  = polynomial.eval y (polynomial.map (eval s) ((fin_succ_equiv R n) f)) :=
+begin
+  apply induction_on f,
+  intro a,
+  simp,
+  intros p q hp hq,
+  simp only [(fin_succ_equiv R n).map_add, ring_hom.map_add, polynomial.map_add,
+             polynomial.eval_add],
+  congr,
+  exact hp,
+  exact hq,
+  intros p j h,
+  simp only [(fin_succ_equiv R n).map_mul, eval_X, polynomial.map_mul, ring_hom.map_mul,
+             polynomial.eval_mul],
+  congr,
+  exact h,
+  clear h f,
+  simp only [fin_succ_equiv_apply, eval₂_hom_X'],
+  by_cases c : j = 0,
+  { rw c,
+    simp [fin.cons_zero] },
+  have c' : j ≠ 0 := by simpa only [ne.def],
+  rw [←fin.succ_pred j c', fin.cons_succ],
+  simp,
+end
+
+lemma coeff_eval_eq_eval_coeff {n : ℕ} {R : Type u} [comm_semiring R] (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) (i : ℕ) :
+  polynomial.coeff (polynomial.map (eval s') f) i =  eval s' (polynomial.coeff f i) :=
+by simp only [polynomial.coeff_map]
+
+/- TODO: what is a good name for this one? -/
+lemma support_eval {n : ℕ} {R : Type u} [comm_semiring R] (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) (i : ℕ) :
+  (polynomial.map (eval s') f).support ⊆ f.support :=
+begin
+  intros i hi,
+  simp only [polynomial.mem_support_iff, polynomial.coeff_map, ne.def] at hi,
+  by_contradiction c,
+  simp only [polynomial.mem_support_iff, not_not, ne.def] at c,
+  rw c at hi,
+  simpa using hi,
+end
+
+lemma degree_eval_le_degree {n : ℕ} {R : Type u} [comm_semiring R] (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) :
+  polynomial.degree (polynomial.map (eval s') f) ≤ polynomial.degree f :=
+begin
+  rw [polynomial.degree, polynomial.degree, finset.sup_le_iff],
+  intros b hb,
+  apply finset.le_sup (support_eval s' f b hb),
+end
+
+lemma nat_degree_eval_le_nat_degree {n : ℕ} {R : Type u} [comm_semiring R] (s : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) :
+polynomial.nat_degree (polynomial.map (eval s) f) ≤ polynomial.nat_degree f
+:= polynomial.nat_degree_le_nat_degree (degree_eval_le_degree s f)
+
+lemma support_coeff_fin_succ_equiv {n : ℕ} {R : Type u} [comm_semiring R]
+  {f : mv_polynomial (fin (n + 1)) R} {i : ℕ} {m : fin n →₀ ℕ } :
+  m ∈ (polynomial.coeff ((fin_succ_equiv R n) f) i).support ↔ (finsupp.cons i m) ∈ f.support :=
+begin
+  apply iff.intro,
+  intro h,
+  simpa [←fin_succ_equiv_coeff_coeff] using h,
+  intro h,
+  simpa [mem_support_iff, ←fin_succ_equiv_coeff_coeff m f i] using h,
+end
+
+lemma fin_succ_equiv_support {R : Type u}[comm_semiring R] {n i : ℕ}
+  {f : mv_polynomial (fin (n + 1)) R} : i ∈ (fin_succ_equiv R n f).support
+  ↔ ∃ m, (finsupp.cons i m) ∈ f.support :=
```

Comment:
I think this statement might be the same as the following more concise statement:
```suggestion
  {f : mv_polynomial (fin (n + 1)) R} :
   (fin_succ_equiv R n f).support = finsupp.cons i '' f.support :=
```
the proof should have to change only slightly (adding an `ext` and `rw mem_image`)

Reply:
I've restated this as 
```(fin_succ_equiv R n f).support = finset.image (λ m : fin (n + 1)→₀ ℕ, m 0) f.support```
and also added another version similar to this:
```finset.image (finsupp.cons i) (polynomial.coeff ((fin_succ_equiv R n) f) i).support  = f.support.filter(λ m, m 0 = i)```
With this new statement the proof of `degree_fin_succ_equiv` became more clean.